### PR TITLE
Reconcile IAM roles API docs into main roles API docs and tag EA

### PR DESCRIPTION
**Pull request by alifatolahi-okta**
## Description:
- Upgrade Beta reference to EA
- Add app resources/permissions
- Reconcile IAM roles API docs into main roles API docs and tag EA (replace the old `roles` content with the contents from `iam-roles` minus a few lines that aren't relevant with EA release now) _**Note that the majority of the content has already been reviewed.**_
- Allow the new content to be searched and indexed
- Monolith Release 2021.08.2

### Resolves:

* [OKTA-394530](https://oktainc.atlassian.net/browse/OKTA-394530)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md [hunk 1]
```diff
@@ -119,10 +139,10 @@ To specify Binding Members, use the REST URL of the corresponding Okta API:
...
 
     As a result, if an admin was granted a standard Role that is limited to a single Group, and at the same time received group management privileges on all Groups in the org through a Custom Role, the ultimate outcome is group management on all Groups.
-2. During BETA, Custom Roles, Permissions and resources are limited to User and Group related items. We plan to introduce additional Resources and Permissions over time and appreciate your feedback.
-3. You can't assign a Custom Role without a Resource Set, hence always being applicable only to a subset of resources. Standard Roles on the other hand, are always initially granted to the entire org. They are only scoped to specific resources by subsequent invoking of the [target operations](/docs/reference/api/roles/#role-target-operations).
+3. <ApiLifecycle access="ea" />You can't assign a Custom Role without a Resource Set, hence always being applicable only to a subset of resources. Standard Roles on the other hand, are always initially granted to the entire org. They are only scoped to specific resources by subsequent invoking of the [target operations](/docs/reference/api/roles/#role-target-operations).
```
Comment (nateyocom-okta) [inline packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md:142]: should this be '2' now?  Also I realize the language didn't change, but maybe we should:  "can't" -> "cannot", and In the second sentence I would suggest changing ", are always initially granted to the entire org" into ", are always scoped to the entire org initially.".

**Review by nateyocom-okta — APPROVED** [review]
Minor comments, LGTM otherwise

**Review by saurabhagarwal-okta — APPROVED** [review]

**Comment by nainikhajanchi-okta** [comment]
@alifatolahi-okta nit: in the permission descriptions, 'User' and 'Group' should be in lowercase as we're referring to users and groups in general.

File: packages/@okta/vuepress-site/docs/reference/api/roles/index.md [hunk 2]
```diff
@@ -1324,48 +2888,437 @@ HTTP/1.1 204 No Content
...
+
+A Custom Role is a custom set of [Permissions](#permission-types). A Custom Role is uniquely identified within your org by its ID or label.
+
+### Custom role properties
+| Property         | Description                         | DataType                                       | Nullable   | Unique   | Read Only |
+| :--------------- | :---------------------------------- | :--------------------------------------------- | :--------- | :------- | :-------- |
+| `id`               | Unique key for the Role             | String                                         | FALSE      | TRUE     | TRUE      |
+| `label`            | Display name of Role                | String                                         | FALSE      | TRUE     | FALSE     |
+| `created`      | The date and time the Role was created | Timestamp | FALSE      | FALSE    | TRUE     |
+| `lastUpdated`      | The date and time the Role's label or description were last updated | Timestamp | FALSE      | FALSE    | TRUE     |
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/roles/index.md:3049]:
```suggestion
| `lastUpdated`      | The date and time that the Role's label or description were last updated | Timestamp | FALSE      | FALSE    | TRUE     |
```

File: packages/@okta/vuepress-site/docs/reference/api/roles/index.md [hunk 3]
```diff
@@ -1324,48 +2888,437 @@ HTTP/1.1 204 No Content
...
+
+<ApiLifecycle access="ea" />
+
+A Custom Role is a custom set of [Permissions](#permission-types). A Custom Role is uniquely identified within your org by its ID or label.
+
+### Custom role properties
+| Property         | Description                         | DataType                                       | Nullable   | Unique   | Read Only |
+| :--------------- | :---------------------------------- | :--------------------------------------------- | :--------- | :------- | :-------- |
+| `id`               | Unique key for the Role             | String                                         | FALSE      | TRUE     | TRUE      |
+| `label`            | Display name of Role                | String                                         | FALSE      | TRUE     | FALSE     |
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/roles/index.md:3047]:
```suggestion
| `label`            | Display name of the Role                | String                                         | FALSE      | TRUE     | FALSE     |
```

File: packages/@okta/vuepress-site/docs/reference/api/roles/index.md [hunk 4]
```diff
@@ -1324,48 +2888,437 @@ HTTP/1.1 204 No Content
...
+Permissions can be added to, or removed from, custom Roles.  Permissions to manage a resource also grant the viewing privileges for the same resource so that you won't need to assign them separately.
+
+User permissions are only effective with respect to the Group(s) to which the admin is granted a Role through Resource Set assignments.
+
+#### Permission properties
+| Property         | Description                         | DataType                                       | Nullable   | Unique   | Read Only |
+| :--------------- | :---------------------------------- | :--------------------------------------------- | :--------- | :------- | :-------- |
+| `label`            | Type of permissions                | String                                         | FALSE      | FALSE     | TRUE     |
+| `created`      | The date and time the Permission was added to the role | Timestamp | FALSE      | FALSE    | TRUE     |
+| `lastUpdated`      | The date and time the permission's association with the role was last updated | Timestamp | FALSE      | FALSE    | TRUE     |
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/roles/index.md:3085]:
```suggestion
| `lastUpdated`      | The date and time the Permission's association with the role was last updated | Timestamp | FALSE      | FALSE    | TRUE     |
```

File: packages/@okta/vuepress-site/docs/reference/api/roles/index.md [hunk 5]
```diff
@@ -1324,48 +2888,437 @@ HTTP/1.1 204 No Content
...
+
+Note that the following fields are different compared to those for [a Group assigned standard Role](#sample-role-assigned-to-the-user-through-a-group-membership):
+
+* The `id` field has a different format, but it has the same application.
+* The `type` field for Custom Roles always has the value `CUSTOM`.
+* The new `resource-set` field gives the ID of the Resource Set to which this assignment applies.
+* The `_links` field has three new fields:
+  * `resource-set` is the `GET` link to the Resource Set to which this assignment is granted.
+  * `role` is the `GET` link to the Role to which this assignment is granted.
+  * `permissions` is the `GET` link to the permissions to which this assignment is granted.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/roles/index.md:2944]:
```suggestion
  * `permissions` is the `GET` link to the Permissions to which this assignment is granted.
```

File: packages/@okta/vuepress-site/docs/reference/api/roles/index.md [hunk 6]
```diff
@@ -1324,48 +2888,437 @@ HTTP/1.1 204 No Content
...
+| `okta.users.manage`                 | Allows the admin to create and manage Users and read all profile and credential information for Users                               | All Users, all Users within a specific Group |
+| `okta.users.create`                 | Allows the admin to create Users. If the admin is also scoped to manage a Group, that admin can add the User to the Group on creation and then manage | All Groups, a specific Group                 |
+| `okta.users.read`                   | Allows the admin to read any User's profile and credential information                                                              | All Users, all Users within a specific Group |
+| `okta.users.credentials.manage`     | Allows the admin to manage only credential lifecycle operations for a User                                                          | All Users, all Users within a specific Group |
+| `okta.users.userprofile.manage`     | Allows the admin to only do operations on the User Object, including hidden and sensitive attributes                                | All Users, all Users within a specific Group |
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/roles/index.md:3095]:
```suggestion
| `okta.users.userprofile.manage`     | Allows the admin to only do operations on the User object, including hidden and sensitive attributes                                | All Users, all Users within a specific Group |
```

**Review by susanharper-okta — COMMENTED** [review]
just some nits.

**Review by susanharper-okta — APPROVED** [review]
LGTM

**Pull request merged**